### ISSUE TITLE
[TD] Addon mixfile loading fails on non-Windows OSes

### ISSUE THREAD

**Issue by hazelnot** [TheAssemblyArmada/Vanilla-Conquer#730]
I can't seem to be able to get VanillaTD to display the Covert Ops missions or skirmish/MP maps. All the files are in place, and I know for a fact it's reading at least some of the files - I have the full soundtrack available, including the 7 CO tracks, and as part of my TD Romanian translation project I discovered that it's reading mission.ini from the CO general.mix file.

I mentioned this on the Discord server, but I thought it might be useful to have this as an issue here for tracking purposes.

Just to clarify in case it's related, I'm running this on Linux.

**Comment by Gerwin2k** [comment]
The mix file setup below works, been using it for more then a year. Covert Ops missions and maps are available in the game. Note that it is a bit of a compacted setup, if you want all music and movies it will be slightly different .
```
Root game folder for VanillaTD
cclocal.mix          06-01-1997     121.305 (From CNC95 install procedure)
deseicnh.mix         19-12-1996     120.296 (From CNC95 install procedure)
local.mix            29-01-1996     114.387 (From Covert Ops CD, name conflict with C&C 95 file)
movies.mix           17-11-2002   3.562.660 (here it is a modded file, just the intro movie, to save space)
sc-000.mix           04-01-1996     185.937 (From Covert Ops CD)
sc-001.mix           02-02-1996     490.318 (From Covert Ops CD)
scores.mix           24-05-1996  39.114.329 (Just using the base game scores mix here, to save space) 
speech.mix           25-08-1995     947.949 (From Covert Ops CD, name conflict with C&C 95 file) 
tempicnh.mix         19-12-1996     119.935 (From CNC95 install procedure)
transit.mix          03-01-1996   4.105.646 (From Covert Ops CD, name conflict with C&C 95 file) 
update.mix           09-01-1997  10.233.756 (From CNC95 install procedure)
updatec.mix          29-07-1996     990.901 (From CNC95 install procedure)
winticnh.mix         19-12-1996     119.935 (From CNC95 install procedure)

SubFolder CovertOps
aud.mix              07-08-1995  1.890.646
conquer.mix          11-01-1996  2.270.373
desert.mix           12-08-1995    677.061
general.mix          15-01-1996  3.409.189
sounds.mix           23-08-1995  2.121.045
temperat.mix         12-08-1995    639.607
winter.mix           12-08-1995    667.385

SubFolder GDI
aud.mix              07-08-1995  1.890.646
conquer.mix          17-05-1996  2.039.435
desert.mix           20-06-1996    677.061
general.mix          10-12-1996  2.584.388
sounds.mix           27-11-1996  1.228.794
temperat.mix         20-06-1996    639.607
winter.mix           20-06-1996    667.385

SubFolder NOD
aud.mix              07-08-1995  1.890.646
conquer.mix          17-05-1996  2.039.435
desert.mix           20-06-1996    677.061
general.mix          11-12-1996  2.549.589
sounds.mix           27-11-1996  1.228.794
temperat.mix         20-06-1996    639.607
winter.mix           20-06-1996    667.385

```

**Comment by hazelnot** [comment]
I already verified with multiple people that I have the right files in the right places. Here's my folder structure:
```
.
├── CCLOCAL.MIX
├── CCLOCAL.MIX.bak
├── CD
│   ├── CD3_Covertops.bin
│   ├── CD3_Covertops.cue
│   ├── CNC95_GDI.iso
│   ├── CNC95_Nod.iso
│   └── README.txt
├── conquer.mix
├── covertops
│   ├── general.mix
│   └── movies.mix
├── DESEICNH.MIX
├── desert.mix
├── gdi
│   ├── general.mix
│   └── movies.mix
├── hpips 0000.png
├── hpips 0001.png
├── hpips 0002.png
├── hpips 0003.png
├── hpips 0004.png
├── hpips 0005.png
├── hpips 0006.png
├── hpips 0007.png
├── hpips.shp
├── nod
│   ├── general.mix
│   └── movies.mix
├── savegame.001
├── savegame.002
├── savegame.003
├── savegame.004
├── sc-000.mix
├── sc-001.mix
├── scores.mix
├── sidebar_edit_palettes.mix
├── sounds.mix
├── speech.mix
├── temperat.mix
├── TEMPICNH.MIX
├── TRANSIT.MIX
├── UPDATEC.MIX
├── UPDATEC.MIX.bak
├── UPDATE.MIX
├── vanillatd
├── winter.mix
└── WINTICNH.MIX
```

The .bak files, PNGs and the extra mix file are there for translation purposes.

I don't remember where each file is from, but I used the instructions [on the wiki](https://github.com/TheAssemblyArmada/Vanilla-Conquer/wiki/Installing-VanillaTD) except I put the CO `scores.mix` in the main folder cause it was overriding the other two anyway (and was working before doing that regardless)

**Comment by Gerwin2k** [comment]
> I already verified with multiple people that I have the right files in the right places. Here's my folder structure:

I see. Well obviously there is some difference that throws a wrench. Since it does not work for you, but you suggest it seems to work for others. Noting the files size / date / source information is no luxury in this. 

There was not a wiki about this back when I set up my installation. If I have some time to spare time I will make another install according to the wiki, to see what it does.

**Comment by Gerwin2k** [comment]
Why does the wiki lack the transit.mix in the final file listing? When I shuffle my listed files around according to the wiki final listing: Covert Ops seems to work but the normal campaign now crashes. 

I don't know how to help any further besides what I wrote already.

**Comment by hazelnot** [comment]
Oh lol, `transit.mix` *is* missing from the final file listing, I'll let @OmniBlade know

Also, weird, I can play the normal campaign without any issue based on the Wiki file listing/table but I'm just not getting the Covert Ops button in the menu

**Comment by Gerwin2k** [comment]
> Also, weird, I can play the normal campaign without any issue based on the Wiki file listing/table but I'm just not getting the Covert Ops button in the menu

I have the 'New Missions' button present in the main menu, even when I remove any Covert Ops related file and folder. Also when I clear out conquer.ini settings. Just that, in that case, the mission listing is empty. It appears with no missions listed there to select and play. This 'New Missions' button and listing is not a covert ops exclusive, since It also lists Playstation or user made missions, when present. 

Are you saying you do not have the 'New Missions' button? Whilst I cannot get rid of it, if I wanted to.

As you may know covert ops missions and maps are packed in sc-000.mix and sc-001.mix. So these two files need to be proper and present the game's root.

**Comment by hazelnot** [comment]
I mean they're in the root folder and I copied them straight from the CO ISO so 😅

But yeah I can't get the New Missions button to appear at all

**Comment by Gerwin2k** [comment]
I tried some more to make the 'New Missions' button to go away. And succeeded by removing my modded placeholder movies.mix file. It contained a file called expand.dat of zero bytes. The presence of expand.dat triggers the 'New Missions' button to appear. As a quick test could you just make an empty file and call it "expand.dat", in the game's root?

**Comment by hazelnot** [comment]
>As a quick test could you just make an empty file and call it "expand.dat", in the game's root?

That... actually did make the New Missions button appear, and it opens an empty menu cause it still can't detect the scenarios, even though the required files *are* there. I also cross-referenced the sizes of `sc-000.mix` and `sc-001.mix` with yours, and they're the exact same size...

Just to clarify in case it's related, I'm running this on Linux.

**Comment by giulianobelinassi** [comment]
Fixed in 3c52d34
